RecoTracker/MkFitCore/src/MkBuilder.cc

```diff
@@ -0,0 +1,1427 @@
+    for (auto &&seed : seeds) {
+      print_seed(seed);
+    }
+  }
+
+  void print_seeds(const EventOfCombCandidates &event_of_comb_cands) {
+    for (int iseed = 0; iseed < event_of_comb_cands.size(); iseed++) {
+      print_seed2(event_of_comb_cands[iseed].front());
+    }
+  }
+#endif
+
+  bool sortCandByScore(const TrackCand &cand1, const TrackCand &cand2) {
+    return mkfit::sortByScoreTrackCand(cand1, cand2);
+  }
+
+}  // end unnamed namespace
+
+//------------------------------------------------------------------------------
+// Constructor and destructor
+//------------------------------------------------------------------------------
+
+namespace mkfit {
+
+  std::unique_ptr<MkBuilder> MkBuilder::make_builder(bool silent) { return std::make_unique<MkBuilder>(silent); }
+
+  void MkBuilder::populate() { g_exe_ctx.populate(Config::numThreadsFinder); }
+
+  //------------------------------------------------------------------------------
+  // Common functions
+  //------------------------------------------------------------------------------
+
+  void MkBuilder::begin_event(MkJob *job, Event *ev, const char *build_type) {
+    m_nan_n_silly_per_layer_count = 0;
+
+    m_job = job;
+    m_event = ev;
+
+    m_seedEtaSeparators.resize(m_job->num_regions());
+    m_seedMinLastLayer.resize(m_job->num_regions());
+    m_seedMaxLastLayer.resize(m_job->num_regions());
+
+    for (int i = 0; i < m_job->num_regions(); ++i) {
+      m_seedEtaSeparators[i] = 0;
+      m_seedMinLastLayer[i] = 9999;
+      m_seedMaxLastLayer[i] = 0;
+    }
+
+    if (!m_silent) {
+      std::cout << "MkBuilder building tracks with '" << build_type << "'"
+                << ", iteration_index=" << job->m_iter_config.m_iteration_index
+                << ", track_algorithm=" << job->m_iter_config.m_track_algorithm << std::endl;
+    }
+  }
+
+  void MkBuilder::end_event() {
+    m_job = nullptr;
+    m_event = nullptr;
+  }
+
+  void MkBuilder::release_memory() {
+    TrackVec tmp;
+    m_tracks.swap(tmp);
+    m_event_of_comb_cands.releaseMemory();
+  }
+
+  void MkBuilder::import_seeds(const TrackVec &in_seeds, std::function<insert_seed_foo> insert_seed) {
+    // bool debug = true;
+
+    const int size = in_seeds.size();
+
+    IterationSeedPartition part(size);
+
+    m_job->m_iter_config.m_partition_seeds(m_job->m_trk_info, in_seeds, m_job->m_event_of_hits, part);
+
+    RadixSort radix;
+    radix.Sort(&part.m_sort_score[0], size);
+
+    for (int i = 0; i < size; ++i) {
+      int j = radix.GetRanks()[i];
+
+      const Track &S = in_seeds[j];
+      HitOnTrack hot = S.getLastHitOnTrack();
+
+      int reg = part.m_region[j];
+
+      ++m_seedEtaSeparators[reg];
+
+      m_seedMinLastLayer[reg] = std::min(m_seedMinLastLayer[reg], hot.layer);
+      m_seedMaxLastLayer[reg] = std::max(m_seedMaxLastLayer[reg], hot.layer);
+
+      // MT-OPTIMIZE-MEM: should delay this and pool-allocate memory for HoTs once
```

Comment:
Could this be moved to an issue?

Reply:
Moved to an internal issue.
Addressed in Round 4 follow-up PR trackreco/cmssw#79 commit: https://github.com/trackreco/cmssw/pull/79/commits/d081786b240bad6e7d367004a8b05c8de61ad18b